src/main/resources/testchipip/csrc/SimUART.cc

```diff
@@ -0,0 +1,44 @@
+#include <vpi_user.h>
+#include <svdpi.h>
+
+#include <stdio.h>
+#include <string.h>
+
+#include "uart.h"
+
+uart_t *uart = 0;
+
+extern "C" void uart_init(
+        const char *filename,
+        int uartno)
+{
+    if (strlen(filename) != 0)
+        uart = new uart_t(filename, uartno);
+    else
+        uart = new uart_t(0, uartno);
+}
+
+extern "C" void uart_tick(
+        unsigned char out_valid,
+        unsigned char *out_ready,
+        char out_bits,
+
+        unsigned char *in_valid,
+        unsigned char in_ready,
+        char *in_bits)
+{
+    if (uart == 0) {
```

Comment:
Might be worth making this guard consistent with the other guards in SimSerial and SimBlockDevice.
Initialize at the top to `NULL` and then here check `!uart` or `uart == NULL` (serial vs blockdevice).

Reply:
Sure. Thats fine.